pytest_asyncio/plugin.py

```diff
@@ -193,10 +198,28 @@ def unused_tcp_port_factory():
 
     def factory():
         """Return an unused port."""
-        port = _unused_tcp_port()
+        port = _unused_port()
+
+        while port in produced:
+            port = _unused_port()
+
+        produced.add(port)
+
+        return port
+    return factory
+
+
+@pytest.fixture
```

Comment:
```suggestion
@pytest.fixture(scope=session)
```
To reflect `unused_tcp_port_factory` fixture definition